tempesta_fw/str.h

```diff
@@ -193,6 +197,46 @@ tfw_str_updlen(TfwStr *s, const char *curr_p)
 	s->len += n;
 }
 
+/**
+ * Returns EOL length
+ */
+static inline int
+tfw_str_eol_len(const TfwStr *s)
+{
+	return s->eol_len;
+}
+
+/**
+ * Updates EOL length value
+ */
+static inline void
+tfw_str_set_eol_len(TfwStr *s, unsigned int len)
+{
+	BUG_ON(len > 2); /* LF and CRLF is the only valid EOL markers */
+	s->eol_len = (unsigned char)len;
+}
+
+/**
+ * Returns total string length, including EOL
+ */
+static inline unsigned long
+tfw_str_total_len(const TfwStr *s)
+{
+	return s->len + s->eol_len;
+}
+
+/**
+ * Converts containig EOL string to string with EOL by cutting of @s
+ * length and setting @eol_len to @len. That function may be usefull
+ * with hand-crafted strings like `Cookie: SID=deadbeef\r\n`.
+ */
+static inline void
+tfw_str_fixup_eol(TfwStr *s, unsigned int len)
```

Comment:
As it stands now, the term `fixup` is used to mean a proper completion of a header field: either closing it when it is complete, or adding a new chunk of data to it. Perhaps, something like `tfw_str_strip_eol()` would be a better name? Or `tfw_str_trim_eol()`, or `tfw_str_conv_eol()`?

Also, this function looks dangerous. I think that a `BUG_ON(s->eol_len)` is justified. The real-world use case is somewhat ambiguous. If the purpose is to convert a `TfwStr{}` with `EOL` to one without `EOL`, then `len` appears to be superficial here, and it's dangerous to use it blindly. Perhaps we should really check that there are `EOL` characters at the end of the string, strip those characters, and set `s->eol_len` accordingly to correct size.

Also, technically speaking, the length of the chunks of `TfwStr{}` should be adjusted as well. That makes the matter more complicated. If `EOL` consists of two characters it may happen that they are in different chunks. If this function is intended only for plain strings, then that should be enforced via a `BUG_ON()`.